glue/core/visual.py

```diff
@@ -25,10 +25,11 @@ class VisualAttributes(HasCallbackProperties):
     :param linestyle: The linestyle (``'solid' | 'dashed' | 'dash-dot' | 'dotted' | 'none'``)
     :param marker: The matplotlib marker shape (``'o' | 's' | '^' | etc``)
     :param markersize: The size of the marker (int)
+    :param preferred_cmap: The colormap required (str)
```

Comment:
I will fix it working.

Reply:
Looking at its implementation in #2131 and your fixes here, it should work just fine with None passed, so really optional.

Reply:
The int here just sets the value, I think it should be a str or a  MPL LinearSegmentedColormap.

Reply:
Yes, misread the code and overlooked the `insinstance(value, str)` conditional – but then what kind of colormap is your `VisualAttributes(preferred_cmap=1)` test example setting?

Reply:
Yes, misread the code and overlooked the `insinstance(value, str)` conditional – but then what kind of colormap is your `VisualAttributes(preferred_cmap=1)` test example setting?

Reply:
The test was just to check it sets the value, as it stands, you could pass it any non-str value and it will just assign it. 

Whether you want that behaviour, I don't know.

Reply:
If it will raise an error later during the data loading as per your original example, I think it should be treated just like an invalid name string. In fact I would rather not silently revert to `None` in such cases, but at least raise a warning – otherwise users may be confused that their inputs are apparently completely ignored. But that may be up for discussion.
In any case this sounds like the only legit inputs here are `mpl.colors.Colormap` instances or valid names thereof.

Reply:
That makes a lot of sense, would you like me to raise an error if the string is an invalid colormap or not either a string or an instance of `mpl.colors.Colormap`?

Reply:
I tend to raising an exception now, since the intended purpose of setting a different `preferred_cmap` will have failed in that case. Pinging @astrofrog for a second look.

Reply:
I have just changed it for now to raise a ValueError in both cases. If you leave it as None, it will set it as None and carry on.